lib/active_record/tenanted/database_tasks.rb

```diff
@@ -103,7 +113,7 @@ def migrate(config)
 
           # dump the schema and schema cache
           if Rails.env.development? || ENV["ARTENANT_SCHEMA_DUMP"].present?
-            if migrated
+            if migrated || !database_already_initialized
```

Comment:
I don't think we need to add the check for `!database_already_initialized`, unless I'm misunderstanding the intent. The tests all pass without it.